src/pages/settings/PasswordPage.js

```diff
@@ -50,48 +50,101 @@ class PasswordPage extends Component {
             currentPassword: '',
             newPassword: '',
             confirmNewPassword: '',
-            isPasswordRequirementsVisible: false,
-            shouldShowPasswordConfirmError: false,
+            errors: {
+                currentPassword: false,
+                newPassword: false,
+                confirmNewPassword: false,
+                confirmPasswordMatch: false,
+            },
         };
 
         this.handleChangePassword = this.handleChangePassword.bind(this);
+        this.getErrorText = this.getErrorText.bind(this);
+        this.isValidPassword = this.isValidPassword.bind(this);
+        this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+        this.clearErrorAndSetValue = this.clearErrorAndSetValue.bind(this);
         this.currentPasswordInputRef = null;
+
+        this.errorKeysMap = {
+            currentPassword: 'passwordPage.errors.currentPassword',
+            confirmNewPassword: 'passwordPage.errors.confirmNewPassword',
+        };
     }
 
     componentWillUnmount() {
         Onyx.merge(ONYXKEYS.ACCOUNT, {error: '', success: ''});
     }
 
-    onBlurNewPassword() {
-        const stateToUpdate = {};
-        if (!this.state.newPassword || !this.isValidPassword()) {
-            stateToUpdate.isPasswordRequirementsVisible = true;
-        } else {
-            stateToUpdate.isPasswordRequirementsVisible = false;
+    /**
+     * Return the error message for the field
+     *
+     * @param {String} field
+     * @returns {String}
+     */
+    getErrorText(field) {
+        if (this.state.errors[field]) {
+            return this.props.translate(this.errorKeysMap[field]);
         }
+        return null;
+    }
 
-        if (this.state.newPassword && this.state.confirmNewPassword && !this.doPasswordsMatch()) {
-            stateToUpdate.shouldShowPasswordConfirmError = true;
-        } else {
-            stateToUpdate.shouldShowPasswordConfirmError = false;
-        }
+    /**
+     * Checks if the password matches regex
+     *
+     * @param {String} password
+     * @returns {Boolean}
+     */
+    isValidPassword(password) {
+        return password.match(CONST.PASSWORD_COMPLEXITY_REGEX_STRING);
+    }
 
-        this.setState(stateToUpdate);
+    /**
+     * Set value for the field in state and clear error flag
+     * @param {String} field
+     * @param {String} value
+     */
+    clearErrorAndSetValue(field, value) {
+        this.setState(prevState => ({
+            [field]: value,
+            errors: {...prevState.errors, [field]: false},
+        }));
     }
 
-    onBlurConfirmPassword() {
-        if ((this.state.newPassword && !this.state.confirmNewPassword) || !this.doPasswordsMatch()) {
-            this.setState({shouldShowPasswordConfirmError: true});
-        } else {
-            this.setState({shouldShowPasswordConfirmError: false});
+    /**
+     * Validate all fields and submit the form if no errors
+     */
+    validateAndSubmitForm() {
```

Comment:
Can we please make one function that validates and another that submits? Here's an example:

https://github.com/Expensify/App/blob/34442f43b644910d61d181e3720af6bef3f17ccb/src/pages/ReimbursementAccount/CompanyStep.js#L173-L181